### ISSUE TITLE
Add nvidia-drm.modeset

### ISSUE THREAD

**Issue by ryzendew** [archlinux/archinstall#2214]
Could nvidia-drm.modeset=1 be added to grub please so we can have Wayland for NVidia out of the box?
but only if the NVidia drivers are selected.

**Comment by Cardiacman13** [comment]
Hi, you can add it in modeprobe.d, this way, it will work with all bootloaders.

**Comment by Torxed** [comment]
> Hi, you can add it in modeprobe.d, this way, it will work with all bootloaders.
> 
> In the same way, you could cancel the gdm rule with: 'ln -s /dev/null /etc/udev/rules.d/61-gdm.rules' to give Nvidia users the option to use Wayland on GNOME.

I like this method, easy to maintain. Any dracks? Early KMS might be the only thing missing but that should be solved elsewhere anyway?

**Comment by Cardiacman13** [comment]
> > Hi, you can add it in modeprobe.d, this way, it will work with all bootloaders.
> > In the same way, you could cancel the gdm rule with: 'ln -s /dev/null /etc/udev/rules.d/61-gdm.rules' to give Nvidia users the option to use Wayland on GNOME.
> 
> I like this method, easy to maintain. Any dracks? Early KMS might be the only thing missing but that should be solved elsewhere anyway?

No, normally there shouldn't be any dracks to put it in modeprobe.d except that people are so used to putting it in the boot-loader options that they might open issues saying it's not done XD . 

Also, normally, early KMS isn't necessary, especially since it makes the initramfs huge, which can cause issues for those using multiple kernels.

**Comment by Torxed** [comment]
> > > Hi, you can add it in modeprobe.d, this way, it will work with all bootloaders.
> > > In the same way, you could cancel the gdm rule with: 'ln -s /dev/null /etc/udev/rules.d/61-gdm.rules' to give Nvidia users the option to use Wayland on GNOME.
> > 
> > 
> > I like this method, easy to maintain. Any dracks? Early KMS might be the only thing missing but that should be solved elsewhere anyway?
> 
> No, normally there shouldn't be any dracks to put it in modeprobe.d except that people are so used to putting it in the boot-loader options that they might open issues saying it's not done XD .
> 
> Also, normally, early KMS isn't necessary, especially since it makes the initramfs huge, which can cause issues for those using multiple kernels.

https://wiki.archlinux.org/title/NVIDIA#Early_loading some users require putting it in early loading, but this can be done post-install. The only thing we need too ensure is that /boot and /EFI are separate to facilitate multiple kernels.

**Comment by Torxed** [comment]
I'm closing this not because it's done, but because #1903 also exists which has the same need.

**Comment by sanchezcarlosjr** [comment]
Unified kernel images could help by indicating the kernel parameters ```nvidia_drm.fbdev=1``` and ```nvidia_drm.modeset=1```, but they must compile the UEFI executables during the installation. Although users must have UEFI, they can work with different bootloaders and load the kernel parameters as soon as possible. For instance, ```mkinitcpio``` builds ```/etc/kernel/cmdline```, which can be used to configure the parameters:
```
root=UUID=0a3407de-014b-458b-b5c1-848e92a327a3 rw quiet bgrt_disable nvidia_drm.modeset=1 nvidia_drm.fbdev=1
```

**Comment by yuceltoluyag** [comment]
note: nvidia-drm.fbdev=1 and nvidia-drm.modeset=1 are now default options, so they are not needed in cmdline anymore:
https://gitlab.archlinux.org/archlinux/packaging/packages/nvidia-utils/-/commit/1b02daa2ccca6a69fa4355fb5a369c2115ec3e22

**Comment by h8d13** [comment]
Nice digging (: we can probably close this
